### DISCUSSION TITLE
Unable to fetch and process images from localhost:8000 🙏

### DISCUSSION THREAD

**Discussion by leamandeep** [vercel/next.js#37530] [Help] [answered]
Hi 👋 Next js Team !! I want help from you guys . I am using Django in back end and next js in front end . When i am sending data from Django to  next js Image component of next cannot read it . It's just showing error.

>Invalid src prop [(http://127.0.0.1:8000/media/src/pexels-scott-webb-1029599.jpg)](http://localhost:3000/(http://127.0.0.1:8000/media/src/pexels-scott-webb-1029599.jpg)) on `next/image`, hostname "127.0.0.1" is not configured under images in your `next.config.js`

How can i fix this problem??
Also ,How can i configure next js for localhost ??

Here are screenshots of my code👇👇
> Custom instance

![image](https://user-images.githubusercontent.com/68345524/172525356-227d752d-84d1-41e1-8ce9-5ef7d1a56682.png)

> using Custom instance inside getstaticsprops

![image](https://user-images.githubusercontent.com/68345524/172525547-0216bed2-46d2-44bf-bf0c-92197e8e3e0f.png)

> just testing image 

![image](https://user-images.githubusercontent.com/68345524/172525739-1264bc1c-472b-439f-bc6a-922845c7f8cc.png)

**Comment by icyJoseph** [comment, marked as answer]
In your case this is a bit of an edge case I think.

Normally, to prevent malicious usage of your site to optimize image for others, Next.js wants you to declare explicitly other domains that'll serve images. In your case, your image server is not resolving to localhost, but to its local IP 127.0.0.1.

One thing you could to, to just move on:
```js
// next.config.js
module.exports = {
  images: {
    domains: ['127.0.0.1'],
  },
}
```

However, I guess when deployed, these two might not be in the same network, you gotta account for that.

I'd also guess that when you return the list of images, the `src` property on them is resolved to 127.0.0.1, and not localhost which is the domain Next.js sees. In a production environment, the Django backend should include the proper domain as part of the src for next/image, and that domain must also be present in `next.config.js`.

**Reply by leamandeep** [reply]
It's working 🥳 . Thanks for help 🙂

**Reply by icyJoseph** [reply]
Awesome! Please, don't forget to label as answered :)

**Reply by buraev** [reply]
man, thanks for help!
